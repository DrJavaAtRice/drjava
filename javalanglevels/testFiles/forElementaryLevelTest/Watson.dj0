abstract class AFriend extends Object {
  abstract String getName();
}
class Watson extends AFriend {
  String getName() {
    return "Watson";
  }
}
