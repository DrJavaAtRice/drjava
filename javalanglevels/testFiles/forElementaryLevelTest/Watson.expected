// Language Level Converter line number map: dj*->java. Entries: 8
//     1->3         2->4         3->5         4->36        5->37        6->38        7->39        8->40   
abstract class AFriend extends Object {
  public abstract String getName();

  /** This method is automatically generated by the Language Level Converter. */
  public AFriend() {
    super();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      AFriend cast = ((AFriend) o);
      return true;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode();
  }
}
class Watson extends AFriend {
  public String getName() {
    return "Watson";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public Watson() {
    super();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      Watson cast = ((Watson) o);
      return true;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode();
  }
}
